optimum/amd/ryzenai/configuration.py

```diff
@@ -20,82 +265,316 @@ class QuantizationConfig:
+
+        self.check_dtype_and_format(self.activations_dtype, "activations_dtype", self.format)
+        self.check_dtype_and_format(self.weights_dtype, "weights_dtype", self.format)
+
+        if self.enable_ipu_cnn:
+            if self.format not in ["qdq"]:
+                raise ValueError('ipu cnn configuration only support format "qdq".')
+            if self.calibration_method not in ["nonoverflow", "mse"]:
+                raise ValueError('ipu cnn configuration only support calibration_method "nonoverflow" and "mse".')
+            if not (self.extra_options.activation_symmetric and self.extra_options.weight_symmetric):
+                raise ValueError(
+                    "ipu cnn configuration requires setting activation_symmetric and weight_symmetric to true."
+                )
+            if self.activations_dtype not in ["uint8", "int8"]:
+                raise ValueError('ipu cnn configuration only support activations_dtype "uint8" and "int8".')
+            if self.weights_dtype not in ["int8"]:
+                raise ValueError('ipu cnn configuration only support weights_dtype "int8".')
+            if self.per_channel:
+                raise ValueError("ipu cnn configuration only supports per tensor.")
+
+    def __setattr__(self, name, value):
+        if name == "extra_options" and isinstance(value, dict):
+            setattr(self, "extra_options", ExtraOptions(**value))
+        else:
+            super().__setattr__(name, value)
+
+    def __getattr__(self, name):
+        value = getattr(self, name)
+        if name == "format":
+            value = self._map_format(value)
+        elif name == "calibration_method":
+            value = self._map_calibration_method(value)
+        elif name in ["activations_dtype", "weights_dtype"]:
+            value = self._map_dtypes(value, name)
+        return value
+
+    def to_diff_dict(self) -> dict:
+        """
+        Returns a dictionary of non-default values in the configuration.
+        """
+        non_default_values = {}
+        for option in fields(self):
+            if option.name == "extra_options":
+                extra_options_dict = getattr(self, option.name).to_diff_dict()
+                if extra_options_dict:
+                    non_default_values[option.name] = extra_options_dict
+            else:
+                value = getattr(self, option.name)
+
+                if value != option.default and value not in ({}, []):
+                    if option.name == "execution_providers" and value == ["CPUExecutionProvider"]:
+                        continue
+
+                    if isinstance(value, Enum):
+                        value = value.name
+                    elif isinstance(value, list):
+                        value = [elem.name if isinstance(elem, Enum) else elem for elem in value]
+
+                    non_default_values[option.name] = value
+        return non_default_values
```

Comment:
To me what would be more interesting is a method to compare two configs.

Reply:
Is the compare method for loading quantization params from `config.json` file?